### DISCUSSION TITLE
Higher bitrate?

### DISCUSSION THREAD

**Discussion by IkkixD** [owncast/owncast#1577] [Ideas]
Why is the bitrate capped to 6000? I just finished setting everything up and get into changing the settings and realized that it won't benefit me as bitrate is capped at 6000 :/

**Comment by gabek** [comment]
There's some discussion around that here: https://github.com/owncast/owncast/issues/1462

**Reply by IkkixD** [reply]
Would you happen to have the information handy on how to actually change it? I don't really have much experience with this and I have no clue what to do there.

**Reply by gabek** [reply]
It's a bit of a broad topic to talk about "how to use APIs", since that's not really an Owncast specific thing. I don't know what your experience is, or what your environment is.

But if I had to come up with the easiest way I can think of:

- Open chrome dev tools
- Change to the Network tab.
- Click "XHR"
- Go to the Owncast video configuration modal.
- Save a video configuration.
- Look for that request in the Chrome dev tools list of requests. (`streamoutputvariants`)
- Right click / Copy as fetch
- Paste it in a text editor
- Change the bitrate to whatever thing you want
- Go back to your browser
- Change from the network tab to the console tab
- Paste the request to run it

Otherwise I'd suggest searching the web for "how to make an api request from" and then where you want to send it from. Windows? Mac terminal? Ruby? If you have curl that's an option but I'm not sure if you do. In short, I'm not sure what would be convenient for you.

**Reply by IkkixD** [reply]
I'll set it back up and try this I suppose.

There should really just be a checkbox that allows for custom values. It really gimps the software IMO to be limited at such low rates. I was so excited to use it until I realized it would be worse than twitch in terms of quality(as I can stream 8000 bitrate there)

**Reply by gabek** [reply]
I completely understand, but as discussed in https://github.com/owncast/owncast/issues/1462, if it's a plain text box that allows for custom values people will type in random, broken, custom values.  There's a balance between being approachable and having sane values for most people, and for allowing power users to do whatever they want.

If you have any suggestions on how to balance these two completely different use cases I'm totally up for hearing your ideas. I obviously want people to be able to use Owncast to do what they want with it. But if I want a good experience for the 80+% of the people who are using a small VPS on a hosting provider, some guardrails need to be installed to keep people from jumping off the ledge.

**Reply by IkkixD** [reply]
I think what I said, having an optional checkbox for a custom value. You can even throw a big warning after you click it.
"UNLESS YOU KNOW EXACTLY WHAT YOU ARE DOING DO NOT CHECK THIS AND ENTER A CUSTOM VALUE" etc.
Or just tuck away an "advanced user" setting that lets you use custom values.

**Reply by gabek** [reply]
I wish that was enough.  Unfortunately it seems that warnings are ignored or not taken seriously. One example is the "video passthrough" setting. You have to go to the advanced options of a video output, turn it on, and then acknowledge that you understand it might cause issues and it gives you a link to learn more.  What I've discovered is nobody reads the warning, and for sure nobody clicks on the link to learn more.  Then when their stream is broken they don't know why and they say Owncast is broken, completely forgetting that they turned on an option that said "this might break things".

So while I'm 100% in agreement with you that the scenario you outlined would be awesome, people have shown that that advanced options and warnings isn't enough.

**Reply by IkkixD** [reply]
I mean, at that point it's their own fault, I just think adding a fair warning is more than enough, otherwise it's just gimping the people who actually want/need the better settings because of the ignorant imo. Or at least have documentation on how to easily change it! I'll be trying what you said shortly, just setting up another vm.

**Reply by IkkixD** [reply]
So, following your advice, anything past 6000 buffers extremely badly for anyone viewing the stream(except on my network.)
It's definitely not my network, as I have 600+ upload, and I can also handle 2 people at 6000- but not one at 8000. Pretty odd, but I decided to uninstall docker(hypervisor on windows/AMD is terrible and interferes with my normal VMs.) and I'll try open streaming platform or maybe a few other options after the rest of my server parts arrive just to rule out 100% that it's something on my end.

As a side note : I also attempted to lower the cpu/gpu usage slider and increase the delay(although there were no usage issues) and it changed nothing.

And the buffering wasn't even really buffering, it was like a few frames skipping, maybe a little audio, and then eventually nothing.

**Reply by gabek** [reply]
>  buffers extremely badly for anyone viewing the stream(except on my network.)

Just to clarify, it works fine for you locally, but viewers outside of your local network have problems?

**Reply by IkkixD** [reply]
> 

Correct. I tried on another pc of mine, even at 20k 120fps, it didn't stress my system at all and played fine on 2 pcs(local network) perfectly. 
While 2 friends of mine tested and had the same exact buffering for a short while and then it just wouldn't load. I lowered it to 10k, same thing. 8k, same thing. 6k, flawless. I've been streaming at 8k on twitch for the last year~ as reference. My Jellyfin server also direct plays media to a few people off my local network(Halfway across the country) and someone is actually currently streaming from it higher than 6k bitrate as well, so I don't think it can be my network at all.(Which is why I'll try a few other applications soon, Owncast just by far seems to be the best in usability otherwise)

**Reply by gabek** [reply]
I guess I'm confused. If you're able to stream from owncast at 20k 120fps locally, then it's working. Owncast doesn't know or care about your internet connection and doesn't make decisions based on it. If you're watching Owncast locally and it's working, then Owncast seems to be working.  Unless I'm misinterpreting what you're saying.

**Reply by IkkixD** [reply]
Nope, you are correct. It doesn't make any sense to me either. But my network can handle 2 clients at 6k, but not 1 at 8k? That's what's confusing. I even stream to twitch currently at 8k. I've streamed to youtube over 10k(although that was before fiber and I had much lower upload.) I've had several people on my jellyfin server(while I was streaming at 8k on twitch) as well.

**Reply by IkkixD** [reply]
Okay. So, this doesn't really make sense to me at all, but I did some more tests. had a friend help me.
6k bitrate
Nobody watching - no network usage
He loaded 1 tab of chrome - 6Mbps+
He loaded a 2nd tab on Edge - 12Mbps+
He loaded a 3rd tab on Edge(incognito) 18-19Mbps+
All running perfectly fine on his end.
So, I got a proxy running on my Laptop(Same network, so proxy is needed to have non local transfer)
Ran another 6 in 3 different browser + incognito modes - 54Mbps as expected.

However, I change it to 7k and it seemed okay for him
8k started to buffer every 15-20seconds
9k starting buffering every few seconds.
10k plays for a few seconds at most and then buffers to infinity.

I'm really confused at all this. Running at 20k bitrate stream runs flawless on local network (so it's not a hardware issue)
I can handle 8 concurrent 6k streams with 0 buffering on any of the videos (3 from him, 5 from my laptop on a proxy)
But as soon as I start increasing it, the stream starts getting really bad really fast. What could possible cause this? Why does running 3 6k streams directly from the same person run perfectly bufferlessly, but a single 10k stream on his end is non stop buffering? It's so confusing. We tried multiple browsers and I had him use my Proxy server too. Same results.

**Reply by IkkixD** [reply]
Just wanted to update the situation. It's something to do with windows.
I tested on 2 other pcs, and it hosted fine there. Ran TCP optimizer on my pc, works perfectly now.
The one who helped me test could watch it fine. 3 other friends all had stutters/laggy streams.
Had all 3 run TCP optimizer (all of them had default settings in windows/network adapter) and all 3 can access it flawlessly now.
I'm not sure what it is, but windows has some fucky settings I guess.

**Reply by gabek** [reply]
Wow, crazy. This is very much outside my depth as I haven't used Windows in many years. It's nuts you have to optimize settings!

**Comment by aperez900907** [comment]
will the project in the roadmap at any point in time develop an adaptative bitrate for low bandwidth and slow latencies clients?

**Reply by gabek** [reply]
Owncast supports adaptive bitrate streaming via HLS. https://owncast.online/docs/video/#how-does-an-owncast-video-stream-work
